lib/listings/useFetchCurrentUserListing.tsx

```diff
@@ -0,0 +1,13 @@
+import { StringFormat } from "firebase/storage";
+import { useQuery } from "react-query";
+import ListingService from "./ListingService";
+import {
+    IBaseListing,
+    IListingAttributes,
+    TListingCollectionReference,
+    TListingDocumentReference,
+  } from "./types"
+
+export default function useFetchCurrentUserListing(userId: TListingCollectionReference) {
```

Comment:
I would just call this hook `useFetchUserListings` it is not synonymous to only the current user. And have the input be named `ref` instead of `userId`